doc/hardware/peripherals/charger.rst

```diff
@@ -5,6 +5,19 @@ Chargers
 
 The charger subsystem exposes an API to uniformly access battery charger devices.
 
+A charger device, or charger peripheral, is a device used to take external power
+provided to the system as an input and provide power as an output downstream to
+the battery pack(s) and system. The charger device can exist as a module, an
+integrated circuit, or as a functional block in a power management integrated
+circuit (PMIC).
+
+The action of charging a battery pack is referred to as a charge cycle. When the
+charge cycle is executed the battery pack is charged according to the charge
+profile configured on the charger device by the host controller. The charge
```

Comment:
question/nit: is the charging profile always configurable? If not this could be slightly reworded I guess

Reply:
Good point. There are some resistor configurable charger devices that lack a control port. We currently do not support them, but it is a future improvement to have in mind.

I will rephrase it.